tests/unit/meta/test_appstream.py

```diff
@@ -749,6 +771,36 @@ def test_appstream_with_multiple_lists(self):
         assert metadata.website == ["https://johnfactotum.github.io/foliate/"]
         assert metadata.donation is None
 
+    def test_appstream_with_malformed_links(self):
+        file_name = "test.appdata.xml"
+        content = textwrap.dedent(
+            """\
+            <?xml version="1.0" encoding="UTF-8"?>
+              <component type="desktop">
+              <id>com.github.maoschanz.drawing</id>
+              <metadata_license>CC0-1.0</metadata_license>
+              <project_license>GPL-3.0-or-later</project_license>
+              <content_rating type="oars-1.1"/>
+              <name>Drawing</name>
+              <url type="homepage">test_at_hello_dot_com</url>
+              <url type="homepage">https://hello.com</url>
+              <url type="vcs-browser">file://test.deb</url>
+              <url type="vcs-browser">http://file_huge</url>
+              <url type="bugtracker">https://github.com/alainm23/planify/issues</url>
+              <url type="bugtracker">https:/helo.com</url>
```

Comment:
Can you encode something into the malformed URLs to make it obvious that they are invalid?  Some of the problems like `:/` vs `://` here are subtle and takes careful reading to understand the expected outcome of the test.


I'm thinking something along the lines of
```suggestion
              <url type="bugtracker">https:/invalid.com</url>
              <url type="bugtracker">https:/bad.com</url>
              <url type="bugtracker">https:/missing-slash.com</url>
```